citations-impl/impl/src/java/org/sakaiproject/citation/impl/CitationListAccessServlet.java

```diff
@@ -494,20 +495,19 @@ private void displayCitations(PrintWriter out, List<Citation> citations, Citatio
 					if (!citation.hasPreferredUrl() ||
 							(citation.hasPreferredUrl() && (!citation.getPreferredUrlId().equals(urlId))))
 					{
-						String urlLabel = null;
+						//need to check customUrl inorder to display correct label as for non-electronic citation customUrl can be empty string
 						try {
-							urlLabel = ( citation.getCustomUrlLabel( urlId ) == null ||
-									citation.getCustomUrlLabel( urlId ).trim().equals("") ) ? rb.getString( "nullUrlLabel.view" ) : Validator.escapeHtml(citation.getCustomUrlLabel(urlId));
-						} catch (IdUnusedException e) {
-							e.printStackTrace();
-						}
+							if (StringUtils.isNotBlank(citation.getCustomUrl(urlId))) {
 
-						try {
-							out.println("\t\t\t\t<a href=\"" + Validator.escapeHtml(citation.getCustomUrl( urlId )) + "\" target=\"_blank\">" + urlLabel + "</a>");
-						} catch (IdUnusedException e) {
+								String urlLabel = (citation.getCustomUrlLabel(urlId) == null ||
```

Comment:
I didnt pay much attention to that bit as it was already there, sure will change.